ulib/libax/src/cbindings/sys.rs

```diff
@@ -0,0 +1,38 @@
+use super::ctypes;
+use crate::debug;
+use axconfig::{PHYS_MEMORY_SIZE, SMP};
+use axerrno::LinuxResult;
+use axhal::mem::PAGE_SIZE_4K;
+use core::ffi::{c_int, c_long};
+
+#[cfg(feature = "alloc")]
+use axalloc::global_allocator;
+
+#[cfg(feature = "alloc")]
+use super::fd_table::FILE_LIMIT;
+#[cfg(not(feature = "alloc"))]
+const FILE_LIMIT: usize = 1024;
+
+/// Return system configuration infomation
+///
+/// Notice: currently only support what unikraft covers
+#[no_mangle]
+pub unsafe extern "C" fn ax_sysconf(name: c_int) -> c_long {
+    debug!("ax_sysconf <= {}", name);
+    ax_call_body!(ax_sysconf, {
+        match name as u32 {
+            // Page size
+            ctypes::_SC_PAGE_SIZE | ctypes::_SC_PAGESIZE => Ok(PAGE_SIZE_4K as c_long),
+            // Total physical pages
+            ctypes::_SC_PHYS_PAGES => Ok((PHYS_MEMORY_SIZE / PAGE_SIZE_4K) as c_long),
+            // Number of processors in use
+            ctypes::_SC_NPROCESSORS_ONLN => Ok(SMP as c_long),
+            // Avaliable physical pages
+            #[cfg(feature = "alloc")]
+            ctypes::_SC_AVPHYS_PAGES => Ok(global_allocator().available_pages() as c_long),
+            // Maximum number of files per process
+            ctypes::_SC_OPEN_MAX => Ok(FILE_LIMIT as c_long),
```

Comment:
Use `super::fd_table::FILE_LIMIT` directly. If `alloc` is not enabled, it should be `0`.